File: src/Neuralm.Presentation.Web/src/messages/responses/GetEnabledTrainingRoomsResponse.ts
import Response from './Response';
import TrainingRoom from '@/models/TrainingRoom';

/**
 * Represents the get enabled training rooms response class.
 */
export default class GetEnabledTrainingRoomsResponse extends Response {
  public trainingRooms: TrainingRoom[];

  /**
   * Initializes a new instance of the GetEnabledTrainingRoomsResponse class.
   */
  constructor(id: string, requestId: string, dateTime: Date, message: string, success: boolean, trainingRooms: TrainingRoom[]) {
    super(id, requestId, dateTime, message, success);
    this.trainingRooms = trainingRooms;
  }
}

File: src/Neuralm.Services/Neuralm.Services.TrainingRoomService/Neuralm.Services.TrainingRoomService.Messages/GetEnabledTrainingRoomsResponse.cs
using Neuralm.Services.Common.Messages.Abstractions;
using Neuralm.Services.TrainingRoomService.Messages.Dtos;
using System;
using System.Collections.Generic;

namespace Neuralm.Services.TrainingRoomService.Messages
{
    /// <summary>
    /// Represents the <see cref="GetEnabledTrainingRoomsResponse"/> class.
    /// </summary>
    public class GetEnabledTrainingRoomsResponse : Response
    {
        /// <summary>
        /// Gets the training rooms. 
        /// </summary>
        public IList<TrainingRoomDto> TrainingRooms { get; set; }

        /// <summary>
        /// Initializes an instance of the <see cref="GetEnabledTrainingRoomsResponse"/> class.
        /// </summary>
        /// <param name="requestId">The request id.</param>
        /// <param name="trainingRooms">The training rooms.</param>
        /// <param name="message">The message.</param>
        /// <param name="success">The success flag.</param>
        public GetEnabledTrainingRoomsResponse(Guid requestId, IList<TrainingRoomDto> trainingRooms, string message = "", bool success = false) : base(requestId, message, success)
        {
            TrainingRooms = trainingRooms;
        }

        /// <summary>
        /// Initializes an instance of the <see cref="GetEnabledTrainingRoomsResponse"/> class.
        /// SERIALIZATION CONSTRUCTOR.
        /// </summary>
        public GetEnabledTrainingRoomsResponse()
        {
            
        }
    }
}
